src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -1288,50 +768,1631 @@ TEST_F(TrajectoryPlannerTest, velocity_trajectory_curved_path_low_acceleration)
+    path_parameters.path                = {.x = {0, 0, 0, 0}, .y = {0, 0, 0, 0}};
+    path_parameters.orientation_profile = {.coefficients = {0, 0, 1, 0}};
+
+    float linear_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    float angular_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    TrajectorySegment_t segment_lengths[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    PositionTrajectoryElement_t trajectory_elements[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+
+    PositionTrajectory_t trajectory;
+    trajectory.trajectory_elements = trajectory_elements;
+
+    trajectory.path_parameters = path_parameters;
+
+    // Get the segment states and the segment lengths
+    app_trajectory_planner_generateStatesAndReturnSegmentLengths(&trajectory,
+                                                                 segment_lengths);
+
+    app_trajectory_planner_getMaxAllowableSpeedProfile(
+        path_parameters.path, path_parameters.num_segments, path_parameters.t_start,
+        path_parameters.t_end, path_parameters.max_allowable_linear_acceleration,
+        max_allowable_speed_profile);
+
+    TrajectoryPlannerGenerationStatus_t status =
+        app_trajectory_planner_createForwardsContinuousLinearSpeedProfile(
+            &trajectory, max_allowable_speed_profile, segment_lengths);
+    EXPECT_EQ(status, OK);
+
+    app_trajectory_planner_createForwardsContinuousAngularSpeedProfile(&trajectory,
+                                                                       segment_lengths);
+
+    status = app_trajectory_planner_modifyTrajectoryToBackwardsContinuous(
+        &trajectory, segment_lengths);
+    EXPECT_EQ(status, OK);
+    status = app_trajectory_planner_modifyTrajectoryToBackwardsContinuous(
+        &trajectory, segment_lengths);
+    EXPECT_EQ(status, OK);
+
+    app_trajectory_planner_generatePositionTrajectoryTimeProfile(&trajectory,
+                                                                 segment_lengths);
+
+    EXPECT_FLOAT_EQ(trajectory.trajectory_elements[0].time, 0.0);
+    EXPECT_NEAR(trajectory.trajectory_elements[1].time, 0.1818, 0.0001);
+    for (unsigned int i = 2; i < path_parameters.num_segments - 1; i++)
+    {
+        EXPECT_EQ(trajectory.trajectory_elements[i].time,
+                  trajectory.trajectory_elements[i - 1].time +
+                      segment_lengths[i - 1].angular_segment_length /
+                          path_parameters.max_allowable_angular_speed);
+    }
+    const float radians_moved_in_acceleration_phase =
+        trajectory.trajectory_elements[path_parameters.num_segments - 1].orientation -
+        trajectory.trajectory_elements[path_parameters.num_segments - 2].orientation;
+
+    EXPECT_NEAR(trajectory.trajectory_elements[path_parameters.num_segments - 1].time,
+                9.1818, 0.0001);
+}
+
+TEST_F(TrajectoryPlannerTest, test_rebalance_trajectory_segment_linear)
```

Comment:
bit confused about the format of this name, should it have a `_limiting` suffix?

Reply:
I changed the name 👍